api/document/js/__tests__/serialize-doc.test.ts

```diff
@@ -53,6 +53,20 @@ describe('serializeDoc()', () => {
     expect(result.title).toEqual('Stufff');
   });
 
+  it('converts footnotes', () => {
+    const node = schema.nodes.para.create({}, [
+      schema.nodes.paraText.create({}, [
+        schema.text('hello'),
+        schema.nodes.inlineFootnote.create({ emblem: '2' }, [
+          schema.text('i am a footnote'),
+        ]),
+      ]),
+    ]);
+
+    const result = serializeDoc(node);
+    expect(result).toMatchSnapshot();
```

Comment:
There were so many different things to test here that I thought it'd be both easier to write and read to simply make it a snapshot test.

Reply:
I generally dislike snapshots as they're difficult to read: not only are they in a separate file, they include _everything_, making it difficult to highlight the important pieces. We may need to add unit tests around this in the future.

Reply:
Hmm, so I'd argue that in certain cases like this one, they're actually _more_ readable than doing a bunch of assertions: in this particular case, the vast majority of the snapshot has to do with the way PM-style footnotes are converted into API-style footnotes (the only "chaff" is the surrounding paragraph).  Also, being able to provide an extremely simple, self-contained example that effectively says "look, if you start with this, it should convert to this" can be more illuminating than making a bunch of assertions about the result that require the reader to work backwards from them to understand what the functionality is actually doing.

That said, I do agree that in a _lot_ of cases, they can have the disadvantages you mentioned.

I also agree that having the snapshots be in a separate file can make them harder to read.  I originally wanted to just copy-paste the JSON representation into an `expect().toEqual()`, but the complaining of the linter led me to avoid that approach.